nni/retiarii/nn/pytorch/api.py

```diff
@@ -312,6 +313,125 @@ def __getitem__(self, item):
         return access
 
 
+class Repeat(nn.Module):
+    """
+    Repeat a block by a variable number of times.
+
+    Parameters
+    ----------
+    blocks : function, list of function, module or list of module
+        The block to be repeated. If not a list, it will be replicated into a list. If a function, it will be called to
+        instantiate a module. Otherwise the module will be deep-copied.
+    depth : int or tuple of int
+        If one number, the block will be repeated by a fixed number of times. If a tuple, it should be (min, max),
+        meaning that the block will be repeated at least `min` times and at most `max` times.
+    """
+    def __init__(self,
+                 blocks: Union[Callable[[], nn.Module], List[Callable[[], nn.Module]], nn.Module, List[nn.Module]],
+                 depth: Union[int, Tuple[int, int]], label=None):
+        super().__init__()
+        self._label = label if label is not None else f'valuechoice_{uid()}'
+        self.min_depth = depth if isinstance(depth, int) else depth[0]
+        self.max_depth = depth if isinstance(depth, int) else depth[1]
+        assert self.max_depth >= self.min_depth > 0
+        if not isinstance(blocks, list):
```

Comment:
what is the meaning of "list of module"? users replicate module to `max_depth` by themselves?

Reply:
Yes